x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/actions/duplicate_exceptions.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ExceptionListClient } from '@kbn/lists-plugin/server';
+import type { ListArray } from '@kbn/securitysolution-io-ts-list-types';
+import { ExceptionListTypeEnum } from '@kbn/securitysolution-io-ts-list-types';
+
+import type { RuleParams } from '../../../rule_schema';
+
+export const duplicateExceptions = async (
+  ruleId: RuleParams['ruleId'],
+  exceptionLists: RuleParams['exceptionsList'],
+  shouldDuplicate: boolean,
+  exceptionsClient: ExceptionListClient | undefined
```

Comment:
In which case `exceptionsClient` could be undefined?

Reply:
Not sure what case exactly in application, but we get initialsise it in `bulk_actions/route.ts` as:

`const exceptionsClient = ctx.lists?.getExceptionListClient();`

So, looks like it can be undefined